src/core/simulation.jl

```diff
@@ -77,9 +71,41 @@ function get_simulation_time(s::Simulation, stage_number::Int64)
     return s.internal.date_ref[stage_number]
 end
 get_ini_cond_chronology(s::Simulation, number::Int64) = get(s.sequence.ini_cond_chronology, s.sequence.order[number], nothing)
+get_name(s::Simulation, stage::Stage) = get(s.sequence.order, get_number(stage), nothing)
 
 
+function _check_sequence(sim::Simulation)
+    key_names = keys(sim.sequence.horizons)
+    for key in key_names
+        resolution = convert(Dates.Minute, PSY.get_forecasts_resolution(PSI.get_sys(sim.stages[key])))
+        horizon = sim.sequence.horizons[key]
+        interval = sim.sequence.intervals[key]
+        horizon_time = resolution * horizon
+        if horizon_time < interval
+            throw(IS.ConflictingInputsError("horizon ($horizon_time) is shorter than interval ($interval) for $key"))
+        end
+    end
+end
 
+function add_cache!(ff::F, sim::Simulation, 
+                            stage::Stage, 
+                            device_model::DeviceModel{I, D}) where {
+                                F<:AbstractAffectFeedForward,
+                                I<:PSY.StaticInjection,
+                                D<:AbstractDeviceFormulation}
```

Comment:
I'd like us to follow this rule:

When a function accepts a parameter whose type is an abstract type and you don't need to use the actual concrete type inside the function, declare the parameter with the abstract type.  Prefer the use of `T` for a parameterized type instead of picking a letter in the type.

```
function add_cache!(
    ff::AbstractAffectFeedForward,
    sim::Simulation, 
    stage::Stage, 
    device_model::DeviceModel{T, AbstractDeviceFormulation}
) where T <: PSY.StaticInjection
```

Reply:
agreed with Dan with the relaxation that the letter could be other things.